frontend/app/mappers/application-history-mapper.server.ts

```diff
@@ -0,0 +1,232 @@
+    if (federalSocialProgramEntity) {
+      insuranceInfo.hasFederalBenefits = true;
+      insuranceInfo.federalSocialProgram = insuranceId.IdentificationID;
+    }
+    const provincialTerritorialSocialProgramEntity = provincialTerritorialSocialPrograms.find((provincialTerritorialSocialInsurance) => provincialTerritorialSocialInsurance.id === insuranceId.IdentificationID);
+    if (provincialTerritorialSocialProgramEntity) {
+      insuranceInfo.hasProvincialTerritorialBenefits = true;
+      insuranceInfo.provincialTerritorialSocialProgram = insuranceId.IdentificationID;
+      insuranceInfo.province = provincialTerritorialSocialProgramEntity.provinceTerritoryStateId;
+    }
+  });
+
+  return insuranceInfo;
+}
+
+function toApplicantInformation(maritalStatus: string, firstName: string | undefined, lastName: string | undefined, sin: string) {
+  return {
+    maritalStatus: maritalStatus,
+    firstname: firstName,
+    lastName: lastName,
+    socialInsuranceNumber: sin,
+  };
+}
+
+function toCommunicationPreferences(emailAddressID: string | undefined, preferredLanguage: string | undefined, preferredMethod: string) {
+  return {
+    email: emailAddressID,
+    preferredLanguage: preferredLanguage,
+    preferredMethod: preferredMethod,
+  };
+}
+
+interface ToChildrenResult {
+  information: { isParent: boolean | undefined; firstName: string | undefined; lastName: string | undefined; dateOfBirth: string | undefined; hasSocialInsuranceNumber: boolean | undefined; socialInsuranceNumber: string | undefined };
+  dentalInsurance: boolean | undefined;
+  dentalBenefits:
+    | {
+        hasFederalBenefits: boolean | undefined;
+        federalSocialProgram: string | undefined;
+        hasProvincialTerritorialBenefits: boolean;
+        provincialTerritorialSocialProgram: string | undefined;
+        province: string | undefined;
+      }
+    | undefined;
+}
+
+function toChildren(relatedPeople: ToRelatedPersonArgs[]) {
+  const children: ToChildrenResult[] = [];
+
+  relatedPeople.forEach((child) => {
+    const dentalBenefits = toDentalBenefits(child.ApplicantDetail.InsurancePlan?.at(0));
+    children.push({
+      information: {
+        isParent: child.ApplicantDetail.AttestParentOrGuardianIndicator,
+        firstName: child.PersonName.at(0)?.PersonGivenName.at(0),
+        lastName: child.PersonName.at(0)?.PersonSurName,
+        dateOfBirth: child.PersonBirthDate.dateTime,
+        socialInsuranceNumber: child.PersonSINIdentification.IdentificationID,
+        hasSocialInsuranceNumber: child.PersonSINIdentification.IdentificationID ? true : false,
+      },
+      dentalInsurance: child.ApplicantDetail.PrivateDentalInsuranceIndicator,
+      dentalBenefits: dentalBenefits,
+    });
+  });
+  return children;
+}
+
+function toPartnerInformation(relatedPerson: ToRelatedPersonArgs | undefined): ToPartnerResult {
```

Comment:
Optional chaining provides more flexibility as it allows easier handling of missing arguments.

(Note that this comment can be applied to other function and interface declarations throughout this file.

```suggestion
function toPartnerInformation(relatedPerson?: ToRelatedPersonArgs): ToPartnerResult {
```

Reply:
Fixed: a3363de42a431757737fa4c338f8126b2aca30e5